src/PDDocImpl.jl

```diff
@@ -124,6 +124,36 @@ end
 get_pd_font!(doc::PDDocImpl, cosfont::CosObject) =
     get!(doc.fonts, cosfont, PDFont(doc, cosfont))
 
-get_pd_xobject!(doc::PDDocImpl, cosxobj::CosObject) = 
+get_pd_xobject!(doc::PDDocImpl, cosxobj::CosObject) =
     get!(doc.xobjs, cosxobj, createPDXObject(doc, cosxobj))
 
+function iterate_treenode!(pgmap::Dict{CosIndirectObjectRef, Int}, node::CosIndirectObject, currpageno::Int)::Int
+    mytype = get(node, cn"Type")
```

Comment:
I am not sure why additional page enumeration logic is built here. Page enumeration can be carried out by `pdDocGetPage`.

Reply:
This function is internal and used by `get_pageref_to_pageno_map()`. I need to have a map PageRef => PageNo which I did not found in current library code, so I write the function to get this.
`pdDocGetPage` seems to work in little opposite direction: having page no returns page object and I also I do not need to load full page - just page number. 
Another idea - maybe for later (it is not in this patch) is to have this map returned by `get_pageref_to_pageno_map()` be stored persistently (e.g. in doc object) and initialized in lazy way.

Reply:
Outlines have a pointer to a destination which looks like this. `[page /XYZ left top zoom]` 
In that case, only having an absolute page index should be fine. Why do you need a mapping from indirect page object reference to page number? Does not look intuitive to me from the PDF spec.

Reply:
> I will try to check why test failed on other architectures.
> At first look there is some compatibility issues with `string() ` function or some differences in how PDF data are being read. One of the issues seems to be related to casting of CDDate type - not related to present changes - probably because of some latest change in Julia head.
> Honestly I'm little disappointed with Julia being such not portable.
> My work and dev system is 64bit Linux and all works perfectly there so I did not expected any problems.
> 
> We have a Christmas Holidays now, so I will be off-line for a while, Will return to this subject near or just after new year.
> If you'd came to any ideas in a meantime - please let me know.
> PS: Can submit separate PR for issue in PDFontMetrics.jl but probably the same test with CDDate will fail again, so I think at first we have to sort out those compatibility issues. Also this fix must be contained in Outline PR because one of unit tests relay on it.

I think I have fixed the issue already. I ensured all the `show` method points to the date and time definition as PDF spec defines. That will not change across releases of Julia. Wish you a Merry Christmas and Happy New Year ahead.

Reply:
I have merged your last commits in master to pdDocGetOutline branch in my fork repo, but it did not pushed here and did not kicked off CI tests. Little bit stacked on that - probably "Resolve Conflicts" button below should be pressed - but it looks like I have no access rights to do this here.

Reply:
PS: Many thanks for your wishes. I also wish You all the best in new Year !